src/toolbar/Toolbar.jsx

```diff
@@ -1,15 +1,256 @@
+                displayArrowRight = (
+                    <div className="chamel-toolbar-right-arrow">
+                        <FontIcon onClick={this._handleArrowClick.bind(this, 1)}
+                                  className="cfi cfi-chevron-right"/>
+                    </div>
+                );
+            }
+
+            // Modify the toolbar main container's style so we can properly display the toolbar icons
+            let styleToolbarMainCon = {
+                float: 'left',
+                width: this.state.toolbarContainerWidth + 'px'
+            };
+
+            return (
+                <div ref="chamelToolbar" className="chamel-toolbar" style={{padding: '0px'}}>
+                    {displayArrowLeft}
+                    <div className="chamel-toolbar-group" style={styleToolbarMainCon}>
+                        {displayIcons}
+                    </div>
+                    {displayArrowRight}
+                </div>
+            );
+        } else {
+
+            // If the toolbar main container is wide enough to display all the icons, then just display the props.children
+            return (
+                <div ref="chamelToolbar" className="chamel-toolbar">
+                    {this.props.children}
+                </div>
+            );
+        }
+    },
+
+    /**
+     * Function that will get all the possible toolbar icons
+     *
+     * @param {React.Children} children The children of the react component that contains the possible toolbar icons
+     * @private
+     */
+    _getToolbarIcons: function (children) {
+
+        // Map thru the react children and evaluate each element if it is a toolbar icon or a parent component
+        React.Children.map(children, function (element) {
+
+            // Make sure that the element we evaluating is an object
+            if(typeof element === 'object') {
+
+                // Evaluate the element type if it is a toolbar icon
+                switch (element.type.displayName) {
+                    case 'FontIcon':
+                    case 'DropDownIcon':
+                    case 'FlatButton':
+                    case 'IconButton':
+                    case 'EnhancedButton':
+                        let ref = toolbarIcons.length;
+                        let icon = React.cloneElement(element, {ref: ref, key: ref});
+                        toolbarIcons[ref] = {
+                            icon: icon,
+                            width: DEFAULTICONWIDTH
+                        }
+                        break;
+
+                    default:
+
+                        // If the element has a children, then let's loop again and find the possible toolbar icons
+                        if (element.props.children) {
+                            this._getToolbarIcons(element.props.children);
+                        }
+                        break;
+                }
+            }
```

Comment:
This makes me uneasy because it means we pretty much need to make sure we update this function any time we add any new elements to the toolbar. It could make for some odd bugs when future programmers are using this. I also don't love having every element icon have a ref and cloning ever element is also a bit odd....